chain/consensus/mir/mine.go

```diff
@@ -1,76 +1,159 @@
 package mir
 
 import (
+	"bytes"
 	"context"
 	"fmt"
 	"time"
 
+	"go.uber.org/zap/buffer"
+
 	"github.com/filecoin-project/go-address"
 	lapi "github.com/filecoin-project/lotus/api"
 	"github.com/filecoin-project/lotus/api/v1api"
 	"github.com/filecoin-project/lotus/chain/consensus/platform/logging"
 	"github.com/filecoin-project/lotus/chain/types"
 	ltypes "github.com/filecoin-project/lotus/chain/types"
+	mirproto "github.com/filecoin-project/mir/pkg/pb/requestpb"
 )
 
-// Mine implements "block mining" using Mir framework.
+// Mine implements "block mining" using the Mir framework.
 //
-// Mine implements the following algorithm:
+// Mine implements the following main algorithm:
 // 1. Retrieve messages and cross-messages from the mempool.
-//    Note, that messages can be added into mempool via the libp2p mechanism and the CLI.
+//    Note, that messages can be added into mempool via the libp2p and CLI mechanism.
 // 2. Send messages and cross messages to the Mir node through the request pool implementing FIFO.
 // 3. Receive ordered messages from the Mir node and parse them.
-// 4. Create the next Filecoin block. Note, only a leader Eudico node, chosen by round-robin election, creates a block.
+// 4. Create the next Filecoin block. Note, only the leader Eudico node, chosen by round-robin election,
+//    creates a block and gets a reward. This simulation is used to conform the Filecoin consensus.
 // 5. Sync this block without sending it over the libp2p network.
```

Comment:
What if a node is a full node but not a validator? How it is informed about new blocks if we don't send it through the libp2p network?

Reply:
I thought that the Sync protocol can be used for that.
If not then we must address this in the next iteration.

Reply:
The Sync protocol handles that as long as it is notified that there is a new block (our current view is below the heaviest height), which I don't think is the case in the current implementations. Have you tried to sync a node that is not a validator with Mir?

I'm fine addressing this (if @matejpavlovic is OK with it) in the next PR, but let's track it in an issue if you don't mind so we don't forget :) )

Reply:
@matejpavlovic, we are only missing your input here, and LGTM to merge. Thanks!

Reply:
At this stage, when using Mir, there is no such thing as a non-validator node yet. Every node in a mir subnet is a validator. Adding support for non-validating nodes has been discussed as a future extension, but it's not on our plate now.

Reply:
@adlrocha 
> Have you tried to sync a node that is not a validator with Mir?

A new node joining a subnet is syncing with current nodes. Then it starts producing new blocks by yourself (if the node is miner). If it not a miner then is stops syncing.